src/libostree/ostree-repo-commit.c

```diff
@@ -678,12 +678,13 @@ create_regular_tmpfile_linkable_with_content (OstreeRepo *self,
        *    e.g. 10 bytes but is actually gigabytes.
        *  - Due to GLib bugs that pointlessly calls `poll()` on the output fd for every write
        */
-      char buf[8192];
+      gsize buf_size = 1048576;
```

Comment:
Might as well `gsize buf_size = MIN(length, 1048576);`

Reply:
I guess that would be fine. As per I'm reading documentation right, with zero-sized files buf would then be NULL and the loop would abort right away.